### PR TITLE
Add some instructions in installing sample-controller on Kubernetes (minikube)

### PR DESCRIPTION
**Pull request by SataQiu**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Add some instructions in installing sample-controller on Kubernetes (minikube)

**Which issue(s) this PR fixes**:
Fixes #kubernetes/sample-controller#24

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/sample-controller/artifacts/examples/rc.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+apiVersion: v1
+kind: ReplicationController
```
Comment (caesarxuchao) [inline staging/src/k8s.io/sample-controller/artifacts/examples/rc.yaml]: Please use a Deployment instead.

File: staging/src/k8s.io/sample-controller/docs/minikube-walkthrough.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+## Build the container image
+
+Using the binary we just built, we will now create a Docker image and push it to our Dockerhub registry so that we deploy it to our cluster.
+There is a sample ```Dockerfile``` located in ```artifacts/simple-image``` we will use this to build our own image.
+
+Again from the root of this repo run the following commands:
+```
+cp ./sample-controller ./artifacts/simple-image/kube-sample-controller
+docker build -t <YOUR_DOCKERHUB_USER>/kube-sample-controller:latest ./artifacts/simple-image
+docker push <YOUR_DOCKERHUB_USER>/kube-sample-controller
```
Comment (caesarxuchao) [inline staging/src/k8s.io/sample-controller/docs/minikube-walkthrough.md]: minikube is running locally, right? If so, we can skip the push, and hard code the registry name.

File: staging/src/k8s.io/sample-controller/docs/minikube-walkthrough.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+...
+```
+
+Save this file and we are then ready to deploy and try out the sample controller.
+
+## Deploy to Minikube
+
+We will need to create several objects in order to setup the sample controller so you will need to ensure you have the ```kubectl``` tool installed. [Install kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/).
+
+```
```
Comment (caesarxuchao) [inline staging/src/k8s.io/sample-controller/docs/minikube-walkthrough.md]: Can we just do a `kubectl apply -f ./`?

If we care about the order, we can merge the order sensitive objects in one yaml file and separate them by `===`.

File: staging/src/k8s.io/sample-controller/artifacts/examples/rc.yaml (hunk 1 above)
Comment (SataQiu) [reply staging/src/k8s.io/sample-controller/artifacts/examples/rc.yaml]: Thank you. I have modified it.

File: staging/src/k8s.io/sample-controller/docs/minikube-walkthrough.md (hunk 2 above)
Comment (SataQiu) [reply staging/src/k8s.io/sample-controller/docs/minikube-walkthrough.md]: The process has been simplified.

File: staging/src/k8s.io/sample-controller/docs/minikube-walkthrough.md (hunk 3 above)
Comment (SataQiu) [reply staging/src/k8s.io/sample-controller/docs/minikube-walkthrough.md]: I have merged them into one yaml `sample-controller.yaml`

**Comment by SataQiu** [comment]
Could you please help to check whether this PR is valuable ?
/assign @fejta

**Comment by fejta** [comment]
@nikhita @munnerz you two seem the most involved with this sample. Where is the best place to document how to use it?

**Pull request closed without merging**